tslint.json

```diff
@@ -0,0 +1,100 @@
+{
+    "defaultSeverity": "error",
+    "extends": "typestrict",
+    "jsRules": {},
+    "rules": {
+        "arrow-parens": true,
+        "ban-ts-ignore": true,
+        "import-spacing": true,
+        "indent": {
+            "options": [
+                "spaces",
+                2
```

Comment:
2 space indentation seems to be a trend with JavaScript libraries that I started noticing around 2 years ago. I was resistant at first but ultimately prefer to go with the community and used to it now. The community justification seems to be that you hit line length limits less often.

Reply:
Let's keep indentation at 4 for now, since that's what's already used in the project.

In theory, using tabs would solve this issue too, since you could just set your own tab width in your editor. But if everyone uses different settings, it can still cause odd alignment issues.

I figure, if your code is so deeply nested that it hits the line length limit, you need to decompose your code. I've also taken a liking spacing code horizontally, e.g. each function parameter on its own line. That helps with version control, which is good at identifying & "blaming" lines changed, less so for anything smaller.

Reply:
I agree that tabs are the logical choice for the very reason you stated (and more). 👌 I wish all developers could see that. But I've realised it's futile trying to fight the tide of developers who absolutely know spaces are the correct indentation to use because it's "the default in every editor".

> each function parameter on its own line
Yeah, I've been doing this more and more lately too.

Will flick it back to 4 spaces.